crates/fluvio-protocol/src/record/batch.rs

```diff
@@ -369,19 +370,15 @@ where
         } else {
             self.batch_len as usize - BATCH_HEADER_SIZE
         };
+        trace!(rec_len, "decoding batch records with len");
+        // not checking remaining bytes, because we do it in the record set
```

Comment:
Need unit test here to check batch header and api contracts are satisfied such as end offset